compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -550,7 +550,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
   def canDecompose(tp: Type): Boolean = {
     val dealiasedTp = tp.dealias
     val res =
-      (tp.classSymbol.is(Sealed) &&
+      (tp.classSymbol.isEffectivelySealed &&
         tp.classSymbol.isOneOf(AbstractOrTrait) &&
```

Comment:
@liufengyun I wonder why we have the restriction `tp.classSymbol.isOneOf(AbstractOrTrait)` here. It should be possible to check concrete sealed classes for exhaustivity, no?